Tests/BugsnagTests.m

```diff
@@ -0,0 +1,80 @@
+//
+//  BugsnagTests.m
+//  Tests
+//
+//  Created by Robin Macharg on 04/02/2020.
+//  Copyright © 2020 Bugsnag. All rights reserved.
+//
+// Unit tests of global Bugsnag behaviour
+
+#import "Bugsnag.h"
+#import "BugsnagTestConstants.h"
+#import <XCTest/XCTest.h>
+
+@interface BugsnagTests : XCTestCase
+
+@end
+
+@implementation BugsnagTests
+
+/**
+ * Test that global metadata is added correctly, applied to each event, and
+ * deleted appropriately.
```

Comment:
When testing events we should probably also check that any alteration to metadata on the event doesn't affect the metadata on the configuration object either. I'm assuming we have existing test coverage for this or can add some in when the changes on `BugsnagEvent` are implemented.

Reply:
Agreed.  Test expanded to cover this case.